lib/model/message.dart

```diff
@@ -53,15 +53,18 @@ mixin MessageStore {
   /// and the update-message event hasn't arrived.
   bool? getEditMessageErrorStatus(int messageId);
 
-  /// Edit a message's content, via a request to the server.
+  /// Makes an edit-message request and starts an edit-outbox lifecycle.
   ///
   /// Should only be called when there is no current edit request for [messageId],
   /// i.e., [getEditMessageErrorStatus] returns null for [messageId].
   ///
+  /// The returned [Future] resolves or rejects with the edit-message request,
+  /// irrespective of when the edit-message event arrives (if it does).
```

Comment:
I think the actual behavior is good, but this description doesn't quite capture it:
* If the request fails, but only after the event has already arrived, the future resolves rather than rejects.
* Similarly if the request fails but the message has been deleted.
* As of this commit, the future actually never rejects (short of a bug, like reaching the `throw StateError` line); that only happens when the `rethrow` is added in a later commit. Probably that rethrow can just be part of this commit.

Reply:
Indeed, thanks for catching this.